conda_build/api.py

```diff
@@ -378,22 +378,28 @@ def create_metapackage(name, version, entry_points=(), build_string=None, build_
 
 def update_index(dir_paths, config=None, force=False, check_md5=False, remove=False, channel_name=None,
                  subdir=None, threads=None, patch_generator=None, verbose=False, progress=False,
-                 hotfix_source_repo=None, convert_if_not_present=False, **kwargs):
+                 hotfix_source_repo=None, shared_format_cache=True, current_index_versions=None, **kwargs):
+    import yaml
     from locale import getpreferredencoding
     import os
-    from .conda_interface import PY3
+    from .conda_interface import PY3, string_types
     from conda_build.index import update_index
     from conda_build.utils import ensure_list
     dir_paths = [os.path.abspath(path) for path in _ensure_list(dir_paths)]
     # Don't use byte strings in Python 2
     if not PY3:
         dir_paths = [path.decode(getpreferredencoding()) for path in dir_paths]
 
+    if current_index_versions and isinstance(current_index_versions, string_types):
```

Comment:
Is the first condition needed? Unless I am missing something, `isinstance()` should on `NoneType` too.

Reply:
I think you're right